pkg/controller/servicebinding/secret.go

```diff
@@ -10,98 +12,108 @@ import (
 	"k8s.io/client-go/dynamic"
 
 	"github.com/redhat-developer/service-binding-operator/pkg/converter"
-	"github.com/redhat-developer/service-binding-operator/pkg/log"
 )
 
-// secret represents the data collected by this operator, and later handled as a secret.
-type secret struct {
-	logger *log.Log          // logger instance
-	client dynamic.Interface // Kubernetes API client
-	ns     string
-	name   string
-}
-
 // buildResourceClient creates a resource client to handle corev1/secret resource.
-func (s *secret) buildResourceClient() dynamic.ResourceInterface {
+func buildResourceClient(client dynamic.Interface, namespace string) dynamic.ResourceInterface {
 	gvr := corev1.SchemeGroupVersion.WithResource(secretResource)
-	return s.client.Resource(gvr).Namespace(s.ns)
+	return client.Resource(gvr).Namespace(namespace)
+}
+
+// name is a utility function to get a checksum of the resource data
+func name(data map[string][]byte) (string, error) {
```

Comment:
Before a similar symbol was scoped by the `secret` type, but now it doesn't and the function (which is available to the entire `servicebinding` package) has a very opaque name. Please change it to `buildSecretHash(data)` or something more meaningful instead.